python/lsst/drp/tasks/build_camera.py

```diff
@@ -0,0 +1,778 @@
+            pixToFocalPlane, focalPlaneToTangentPlane = self._split_model(newTPx, newTPy, detectorList)
+
+        # Turn the mappings into a Camera object.
+        camera = self._translate_to_afw(
+            pixToFocalPlane, focalPlaneToTangentPlane, detectorList, inputCamera, rotationAngle
+        )
+
+        return camera
+
+    def _normalize_model(self, mapParams, mapTemplate, detectorList, visitList):
+        """Normalize the camera mappings, such that they correspond to the
+        average visit.
+
+        In order to break degeneracies, `gbdes` sets one visit to have the
+        identity for the per-visit part of its model. This visit is thus
+        imprinted in the per-detector part of the model. This function
+        normalizes the model such that the parameters correspond instead to the
+        average per-visit behavior. This step assumes that the per-visit part
+        of the model is a first-order polynomial, in which case this conversion
+        can be done without loss of information.
+
+        Parameters
+        ----------
+        mapParams : `dict` [`float`]
+            Parameters that describe the per-detector and per-visit parts of
+            the astrometric model.
+        mapTemplate : `dict`
+            Dictionary describing the format of the astrometric model,
+            following the convention in `gbdes`.
+        detectorList : `list` [`int`]
+            List of detector ids.
+        visitList : `list` [`int`]
+            List of ids for visits that were used to train the input model.
+
+        Returns
+        -------
+        newDeviceArray : `np.ndarray`
+            Array of NxM, where N is the number of detectors, and M is the
+            number of coefficients for each per-detector mapping.
+        newTPx, newTPy : `np.ndarray`
+            Projection of `self.x` and `self.y` onto the tangent plane, given
+            the normalized mapping.
+        """
+
+        # Get the per-device and per-visit parts of the model.
+        deviceParams = []
+        visitParams = []
+        for element in mapTemplate["BAND/DEVICE"]["Elements"]:
+            for detector in detectorList:
+                detectorTemplate = element.replace("DEVICE", str(detector))
+                detectorTemplate = detectorTemplate.replace("BAND", ".+")
+                for k, params in mapParams.items():
+                    if re.fullmatch(detectorTemplate, k):
+                        deviceParams.append(params)
+        deviceArray = np.vstack(deviceParams)
+
+        for element in mapTemplate["EXPOSURE"]["Elements"]:
+            for visit in visitList:
+                visitTemplate = element.replace("EXPOSURE", str(visit))
+                for k, params in mapParams.items():
+                    if re.fullmatch(visitTemplate, k):
+                        visitParams.append(params)
+        identityVisitParams = np.array([0, 1, 0, 0, 0, 1])
```

Comment:
For my personal understanding why this is like this?

Reply:
This is an array of coefficients that correspond to the `c` terms in `x' = c_{0} + c_{1} x + c_{2} y` and `y' = c_{3} + c_{4} x + c_{5} y`, meaning that this transform will give `x' = x` and `y' = y`.
In `gbdes`, in order to break degeneracies, one of the visits gets picked to have an Identity map for the per-visit, so I have to add that in here, along with the map parameters that are actually fit.